### PR TITLE
Fix to #18002 - Query/Test: convert AssertQuery methods to strongly typed versions (part2)

### PR DESCRIPTION
**Pull request by maumar**
Converting remaining assert query methods + final cleanup

### CODE REVIEW

File: test/EFCore.Specification.Tests/Query/ComplexNavigationsQueryTestBase.cs [hunk 1]
```diff
@@ -9,19 +9,6 @@
...
-// ReSharper disable PossibleMultipleEnumeration
-// ReSharper disable UnusedVariable
-// ReSharper disable EqualExpressionComparison
-// ReSharper disable AccessToDisposedClosure
-// ReSharper disable StringStartsWithIsCultureSpecific
-// ReSharper disable InconsistentNaming
-// ReSharper disable MergeConditionalExpression
-// ReSharper disable ReplaceWithSingleCallToSingle
-// ReSharper disable ReturnValueOfPureMethodIsNotUsed
-// ReSharper disable ConvertToExpressionBodyWhenPossible
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/Query/ComplexNavigationsQueryTestBase.cs]: Why did you remove the suppressions?

Comment (maumar) [reply test/EFCore.Specification.Tests/Query/ComplexNavigationsQueryTestBase.cs]: will bring it back. thought nobody is using it anymore
